timeseries/src/autogluon/timeseries/models/multi_window/multi_window_model.py

```diff
@@ -0,0 +1,190 @@
+                window_idx=window_idx,
+                suffix=f"_W{window_idx}",
+            )
+
+            logger.debug(f"\tWindow {window_idx}")
+            model = self.get_child_model(window_idx)
+            model_fit_start_time = time.time()
+            model.fit(
+                train_data=train_fold,
+                val_data=val_fold,
+                time_limit=None if time_limit is None else time_limit - (model_fit_start_time - global_fit_start_time),
+                **kwargs,
+            )
+            model.fit_time = time.time() - model_fit_start_time
+            model.score_and_cache_oof(val_fold, store_val_score=True, store_predict_time=True)
+            trained_models.append(model)
+
+            logger.debug(f"\t\t{model.val_score:<7.4f}".ljust(15) + f"= Validation score ({model.eval_metric})")
+            logger.debug(f"\t\t{model.fit_time:<7.3f} s".ljust(15) + "= Training runtime")
+            logger.debug(f"\t\t{model.predict_time:<7.3f} s".ljust(15) + "= Training runtime")
+
+            self.info_per_val_window.append(
+                {
+                    "window_idx": window_idx,
+                    "fit_time": model.fit_time,
+                    "val_score": model.val_score,
+                    "predict_time": model.predict_time,
+                }
+            )
+
+        # Only the model trained on most recent data is saved & used for prediction
+        self.most_recent_model = trained_models[0]
+        self.most_recent_model_path = trained_models[0].path
+        self.predict_time = self.most_recent_model.predict_time
+        self.fit_time = time.time() - global_fit_start_time - self.predict_time
+        self._oof_predictions = pd.concat([model.get_oof_predictions() for model in trained_models])
+        self.val_score = np.mean([model.val_score for model in trained_models])
+
+    def get_info(self) -> dict:
+        info = super().get_info()
+        info["info_per_val_window"] = self.info_per_val_window
+        return info
+
+    def get_child_model(self, window_idx: int) -> AbstractTimeSeriesModel:
+        model = copy.deepcopy(self.model_base)
+        model.set_contexts(self.path + f"W{window_idx}" + os.sep)
+        return model
+
+    def predict(
+        self,
+        data: TimeSeriesDataFrame,
+        known_covariates: Optional[TimeSeriesDataFrame] = None,
+        **kwargs,
+    ) -> TimeSeriesDataFrame:
+        if self.most_recent_model is None:
+            raise ValueError(f"{self.name} must be fit before predicting")
+        return self.most_recent_model.predict(data, known_covariates=known_covariates, **kwargs)
+
+    def score_and_cache_oof(
+        self,
+        val_data: TimeSeriesDataFrame,
+        store_val_score: bool = False,
+        store_predict_time: bool = False,
+    ) -> None:
+        # self.val_score, self.predict_time, self._oof_predictions already saved during _fit()
+        assert self._oof_predictions is not None
```

Comment:
Agreed, I removed the assertion from the parent (`AbstractTimeSeriesModel`) but kept it here.

This will produce the same effect for both `MultiWindowBacktestingModel` and any other model:
- multi-window case: assertion will trigger if the model hasn't been fit before
- usual model: assertion will trigger because the model cannot predict (since it hasn't been fit yet)